### PR TITLE
add house deletion endpoint

### PR DESCRIPTION
**Pull request by ike-tom**

### CODE REVIEW

File: src/controllers/house.controllers.js [hunk 1]
```diff
@@ -11,5 +11,12 @@ const createNewHouseController = async (houseData) => {
...
 };
-
+export const getById = async (id) => {
+  try {
+    const favorite = await favoriteModel.findById(id);
+    return { status: 'success', favorite };
+  } catch (err) {
+    return { status: 'invalid', message: 'There is no favorite with this ID' };
+  }
+};
```
Comment (mgoniszewska) [inline src/controllers/house.controllers.js:21]: The favorite model should be only used by the favorite controller and then if you want to use it you should import the necessary function from the favorite controller.

Comment (mgoniszewska) [reply src/controllers/house.controllers.js:21]: You can use `findByIdAndUpdate()` function.
I think that this task needs the house's status changed and not only removed from favorites.

**Review by mgoniszewska — CHANGES_REQUESTED** [review]

File: src/routes/houses.routes.js [hunk 2]
```diff
@@ -9,6 +11,11 @@ const createNewHouseRoutes = (router) => {
     if (response.status === 'invalid') return res.status(StatusCodes.BAD_REQUEST).json(response);
     res.status(StatusCodes.CREATED).json(response);
   });
+  router.delete('/houses/:houseId', async (req, res) => {
```
Comment (FilipHalon) [inline src/routes/houses.routes.js:14]: Isn't this endpoint currently returning a single favorite instance?

**Review by FilipHalon — CHANGES_REQUESTED** [review]

**Review by mgoniszewska — APPROVED** [review]

File: src/routes/houses.routes.js [hunk 3]
```diff
@@ -9,6 +10,11 @@ const createNewHouseRoutes = (router) => {
     if (response.status === 'invalid') return res.status(StatusCodes.BAD_REQUEST).json(response);
     res.status(StatusCodes.CREATED).json(response);
   });
+
+  router.patch('/houses/:id', auth, async (req, res) => {
+    const { id } = req.params;
+    houseDeletion(id);
+  });
```
Comment (FilipHalon) [inline src/routes/houses.routes.js:17]: There's no response.

File: src/controllers/house.controllers.js [hunk 4]
```diff
@@ -13,6 +13,22 @@ const createNewHouseController = async (houseData) => {
   }
 };
 
+export const deleteHouse = async (_id) => {
+  const house = await House.findOneAndUpdate(
+    {
+      _id,
+      houseStatus: { $not: { $eq: 0 } },
+    },
+    { houseStatus: 0 }
```
Comment (FilipHalon) [inline src/controllers/house.controllers.js:22]: Please use here the corresponding constant.

**Review by FilipHalon — APPROVED** [review]

**Pull request merged**